### PR TITLE
Benji/ios 459 paging experience for long messages

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: iOS/Flows/Conversation/ConversationListCoordinator+Presentation.swift [hunk 1]
```diff
@@ -57,6 +57,10 @@ extension ConversationListCoordinator {
         guard let conversation = self.activeConversation else { return }
         let coordinator = PeopleCoordinator(router: self.router, deepLink: self.deepLink)
         coordinator.selectedConversationCID = self.activeConversation?.cid
+        
+        coordinator.toPresentable().dismissHandlers.append { [unowned self] in
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/ConversationListCoordinator+Presentation.swift:61]: handles cancel

File: iOS/Flows/Conversation/ConversationListCoordinator+Presentation.swift [hunk 2]
```diff
@@ -165,7 +169,7 @@ extension ConversationListCoordinator {
         
         let controller = ChatClient.shared.channelController(for: cid)
         
-        let alertController = UIAlertController(title: "", message: "", preferredStyle: .actionSheet)
+        let alertController = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/ConversationListCoordinator+Presentation.swift:172]: doesn't show a space when nil

File: iOS/Flows/People/PeopleViewController+Search.swift [hunk 3]
```diff
@@ -16,8 +16,12 @@ extension PeopleViewController: UISearchBarDelegate {
     
     func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
         var snapshot = self.dataSource.snapshot()
-        let items: [PeopleCollectionViewDataSource.ItemType] = self.allPeople.map { person in
-            return .person(person)
+        let items: [PeopleCollectionViewDataSource.ItemType] = self.allPeople.sorted().map { person in
```
Comment (Benjamindodgson) [inline iOS/Flows/People/PeopleViewController+Search.swift:19]: Added a sort and custom selection logic as we reset the collection view but want to maintain what items are selected.

File: iOS/Flows/People/PeopleViewController.swift [hunk 4]
```diff
@@ -118,6 +119,24 @@ class PeopleViewController: DiffableCollectionViewController<PeopleCollectionVie
             self.button.top = self.view.height
         }
     }
+    
+    func updateSelectedPeopleItems() {
+         let updatedItems: [PeopleCollectionViewDataSource.ItemType] = self.dataSource.itemIdentifiers(in: .people).compactMap { item in
```
Comment (Benjamindodgson) [inline iOS/Flows/People/PeopleViewController.swift:124]: New selection logic

File: iOS/Models/Contact.swift [hunk 5]
```diff
@@ -10,7 +10,23 @@ import Foundation
 import Contacts
 import PhoneNumberKit
 
-struct Person: Avatar, Hashable {
+struct Person: Avatar, Hashable, Comparable {
+    
+    static func < (lhs: Person, rhs: Person) -> Bool {
```
Comment (Benjamindodgson) [inline iOS/Models/Contact.swift:15]: I think this sort by connection, then contact, but haven't fully tested it.

File: iOS/Models/SendableType.swift [hunk 6]
```diff
@@ -12,7 +12,7 @@ import Intents
 protocol Sendable: AnyObject {
     var kind: MessageKind { get set }
     var context: MessageContext { get set }
-    var emotion: Emotion { get set }
+    var emotion: Emotion? { get set }
```
Comment (Benjamindodgson) [inline iOS/Models/SendableType.swift:15]: I made emotions optional.

File: iOS/UI/Swipeable Input/EmotionView.swift [hunk 7]
```diff
@@ -38,24 +22,29 @@ extension Binding {
 
 struct EmotionView: View {
     
-    let emotion: Emotion
+    @ObservedObject var config: MessageDetailConfig
```
Comment (Benjamindodgson) [inline iOS/UI/Swipeable Input/EmotionView.swift:25]: This fixes issue with the emoji not updating correctly.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**